### PR TITLE
Use kubernetes/client-go

### PR DESCRIPTION
**Pull request by tomdee**

### CODE REVIEW

**Comment by tomdee** [comment]
Fixes #171

File: k8s/k8s.go [hunk 1]
```diff
@@ -162,11 +163,10 @@ func CmdAddK8s(args *skel.CmdArgs, conf utils.NetConf, hostname string, calicoCl
 	return result, nil
 }
 
-func newK8sClient(conf utils.NetConf) (*k8sclient.Client, error) {
+func newK8sClient(conf utils.NetConf, logger *log.Entry) (*kubernetes.Clientset, error) {
 	// Some config can be passed in a kubeconfig file
 	kubeconfig := conf.Kubernetes.Kubeconfig
 
-	// Config can be overridden by config passed in explicitly in the network config.
```
Comment (caseydavenport) [inline k8s/k8s.go]: Why did you remove this comment?

File: k8s/k8s.go [hunk 2]
```diff
@@ -205,10 +205,13 @@ func newK8sClient(conf utils.NetConf) (*k8sclient.Client, error) {
 		return nil, err
 	}
 
-	return k8sclient.New(config)
+	logger.Debugf("Kubernetes config %v", config)
+
+	// creates the clientset
```
Comment (caseydavenport) [inline k8s/k8s.go]: Capitalize "creates" please.

**Comment by caseydavenport** [comment]
We should have an automated test for this, but lets do that in a separate PR.

**Comment by caseydavenport** [comment]
A couple of comment nits, but then LGTM.
